core/src/main/java/org/elasticsearch/index/mapper/NumberFieldMapper.java

```diff
@@ -751,6 +778,12 @@ Number valueForSearch(Number value) {
         }
 
         /**
+         * @throws IllegalArgumentException if value is not finite for this type
+         */
+        void validateParsed(Number value) {
+        }
```

Comment:
I made validation as a part of parse method in the same way as it did in byte/short/int/long. Thanks for suggestion, @rjernst!